.ci/scripts/test_global_patch.py

```diff
@@ -18,52 +18,84 @@
 import subprocess
 import sys
 
-# test patching from command line
-err_code = subprocess.call(
-    [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]
-)
-assert not err_code
-from sklearn.svm import SVC, SVR
-
-assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-from sklearnex import patch_sklearn, unpatch_sklearn
-
-# test unpatching from command line
-err_code = subprocess.call([sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"])
-assert not err_code
-unpatch_sklearn()
-from sklearn.svm import SVC, SVR
-
-assert not SVC.__module__.startswith("daal4py") and not SVC.__module__.startswith(
-    "sklearnex"
-)
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-# test patching from function
-patch_sklearn(name=["svc"], global_patch=True)
-from sklearn.svm import SVC, SVR
-
-assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-# test unpatching from function
-unpatch_sklearn(global_unpatch=True)
-from sklearn.svm import SVC, SVR
-
-assert not SVC.__module__.startswith("daal4py") and not SVC.__module__.startswith(
-    "sklearnex"
-)
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
+import pytest
+
+
+@pytest.fixture
+def patch_from_command_line(request):
+    err_code = subprocess.call(
+        [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]
+    )
+    assert not err_code
+
+    def unpatch_from_cmd():
+        err_code = subprocess.call(
+            [sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"]
+        )
+        assert not err_code
+
+    request.addfinalizer(unpatch_from_cmd)
+    return
+
+
+def test_patching_from_command_line(patch_from_command_line):
+    from sklearn.svm import SVC, SVR
+
+    assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
+    assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
+        "sklearnex"
+    )
+
+
+def test_unpatching_from_command_line(patch_from_command_line):

```

Comment:
why is the fixture used here again? Is this trying to expand functionality or match the previous one (because this is slightly different from what was before).  What was previous assumed that the global setting was returned to normal after the unpatch, so by reapplying the ```"sklearnex.glob", "patch_sklearn", "-a", "svc"``` the state of this test is different than what occurred in the file before.

Reply:
The fixture has a finalizer. So on each test, it gets initialized (global patching is applied), and after the test finishes, gets finalized (global unpatching is applied). Then the same repeats for the next test that uses the fixture, ensuring that state doesn't propagate across tests. This is of course assuming that patching and unpatching are working correctly, which is what the tests are looking at, but I thought it better than the idea of merging patching and unpatching in a single test.